openerp/models.py

```diff
@@ -1741,6 +1741,11 @@ def _name_search(self, cr, user, name='', args=None, operator='ilike', context=N
         if not self._rec_name:
             _logger.warning("Cannot execute name_search, no _rec_name defined on %s", self._name)
         elif not (name == '' and operator == 'ilike'):
+            if operator is None:
+                from openerp.osv import fields
+                column = self._all_columns[self._rec_name].column
+                operator = ('=' if isinstance(column, fields.integer)
+                            else 'ilike')
```

Comment:
Please get rid of the old api!  The ID field has type 'integer' like regular integer fields.

    if operator is None:
        field = self._fields[self._rec_name]
        operator = ('=' if field.type == 'integer' else 'ilike')